build: Don't use QuickTime SDK when compiling Windows engine

peter-b opened the pull request:
In LiveCode 8.1, the engine no longer uses QuickTime, so it's not
necessary to use the SDK when compiling.  This ensures that
QuickTime's bogus not-compatible C headers aren't incorrectly used
when compiling the engine.

QuickTime is still needed for compiling the revVideoGrabber external,
and that's left unchanged.

peter-b committed:
build: Don't use QuickTime SDK when compiling Windows engine

In LiveCode 8.1, the engine no longer uses QuickTime, so it's not
necessary to use the SDK when compiling.  This ensures that
QuickTime's bogus not-compatible C headers aren't incorrectly used
when compiling the engine.

QuickTime is still needed for compiling the revVideoGrabber external,
and that's left unchanged.

livecodefraser commented:
@livecode-vulcan review okay 22e7497

livecode-vulcan commented:
💙 review by @livecodefraser ok 22e7497a6dc2856ce4e1e97eab054830c949db8a

livecode-vulcan commented:
😞 test failure 22e7497a6dc2856ce4e1e97eab054830c949db8a
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2718)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2728)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2713)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2729)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2735)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2741)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2753)

peter-b committed:
build: Include '-lversion' in Windows compile

QTML must have previously included this library (which forwards to
`-lmincore` on Windows 8+), and the removal of QTML from the link
exposed the engine's dependency.

livecodefraser commented:
@livecode-vulcan review okay 0aed088

livecode-vulcan commented:
💙 review by @livecodefraser ok 0aed08864545efd724086e46373c624e2eca436a

livecode-vulcan commented:
😎 test success 0aed08864545efd724086e46373c624e2eca436a
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2719)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2729)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2714)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2730)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2736)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2742)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2754)
